src/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -96,17 +98,36 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 						String body = msg.getResponseBody().toString();
 						for (AppPattern p : app.getHtml()) {
 							if (p.getPattern().matcher(body).find()) {
-								logger.debug("WAPP body matched " + app.getName());
+								LOGGER.debug("WAPP body matched " + app.getName());
 								found = true;
 								break;
 							}
 						}
 					}
 					if (! found) {
+						List<Element> metaElements = source.getAllElements(HTMLElementName.META);
+						for (Map<String, AppPattern> sp : app.getMetas()) {
+							for (Map.Entry<String, AppPattern> entry : sp.entrySet()) {
+								for (Element metaElement : metaElements) {
+									String name = metaElement.getAttributeValue("name");

```

Comment:
This loop could still be the outer loop (to avoid iterate the other collections when it has none meta elements).